lib/supabase/getServerMessages.ts

```diff
@@ -0,0 +1,46 @@
+import { HumanMessage, AIMessage, BaseMessage } from "@langchain/core/messages";
+import supabase from "./serverClient";
+
+/**
+ * Server-side message fetching function that uses the service role key
+ * This function should ONLY be used server-side as it has elevated database privileges
+ * and returns data in LangChain format for AI processing
+ * 
+ * @param roomId The ID of the chat room to fetch messages for
+ * @param limit Maximum number of messages to retrieve (default: 100)
+ * @returns Array of LangChain messages in chronological order
+ */
+export async function getServerMessages(roomId: string, limit = 100): Promise<BaseMessage[]> {
+  try {
+    const { data, error } = await supabase
+      .from("memories")
+      .select("*")
+      .eq("room_id", roomId)
+      .order("updated_at", { ascending: true })
```

Comment:
`ascending: true` will give you `earliest` messages first.

My default would be `latest` messages first `ascending: false`

Could you help me understand why you've set `ascending` to `true`?